src/borg/testsuite/efficient_bytes_queue.py

```diff
@@ -0,0 +1,40 @@
+from tempfile import TemporaryFile
+
+import pytest
+
+from ..helpers.datastruct import EfficientBytesQueue
+
+
+class TestEfficientQueue:
+    def test_base_usage(self):
+        sut = EfficientBytesQueue(100)
+        assert not sut.peek_front()
+        sut.push_back(b'1234')
+        assert sut.peek_front() == b'1234'
+        assert len(sut) == 4
+        assert sut
+        sut.pop_front(4)
+        assert not sut.peek_front()
```

Comment:
rather `assert sut.peek_front() == b''` (if the result shall always be the same data type, here `bytes`).

alternatively, `None` could also be an option.